### ISSUE TITLE
networkd sending second RA with different priority + missing OtherInformation

### ISSUE THREAD

**Issue by cooperlees** [systemd/systemd#27982]
### systemd version the issue has been seen with

252.5-2ubuntu3

### Used distribution

Ubuntu 23.04

### Linux kernel version used

6.2.0-20-generic

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

I see two RAs from each of my home routers. One with the information I expect (due to configuration) and one with the default medium preference causing nodes on my home network to ECMP (Equal Cost Multi Path) router to my primary cable and backup 5g internet. I'd like to have everything use Cable and only fail over the 5g if cable fails. This second RA seems to make that difficult as other boxes default routes look like this:

```
default proto ra metric 600 pref high
        nexthop via fe80::2 dev wlan0 weight 1
        nexthop via fe80::3 dev wlan0 weight 1
```
- This might be NetworkManager RA bug/misconfig on one of my raspberry Pi's tho ...

I only expect to see RA's with high preference from home1 and low preference from home2 router.

Here is the **home2** backup router config

```ini
[Match]
Name=vlan69

[Network]
Description=Main user VLAN
DHCP=no
Address=10.6.9.3/24
Address=fd00:1::3/64
Address=fe80::3/64
DNS=2001:4860:4860::6464
DNS=2001:4860:4860::64
DHCPServer=no
IPForward=yes
IPv6SendRA=yes
IPv6AcceptRA=no
LLDP=yes
EmitLLDP=nearest-bridge

[IPv6AcceptRA]
UseDNS=no
RouteMetric=1469
UseGateway=no

# Floating Static routes - metric higher than DHCP learnt default from ISP
[Route]
Destination=0.0.0.0/0
Gateway=10.6.9.2
Metric=1069

[Route]
Destination=::/0
Gateway=fe80::2
Metric=1469

[IPv6Prefix]
Prefix=fd00:1::/64
Assign=yes

[IPv6SendRA]
Managed=no
RouterLifetimeSec=30
RouterPreference=low
OtherInformation=yes
EmitDNS=yes
DNS=fd00:70::69
EmitDomains=yes
Domains=cooperlees.com
DNSLifetimeSec=600
```

Is that a correct assumption that I should only see 1 RA periodically or should we expect the two RAs shared below in the tcpdump output? This system is one of two networkd "routers" I have in the broadcast domain (vlan). They are both networkd with the similar configs.

```
  INTERNET 1                       INTERNET 2
           |                            |
[home1 router] --- VLAN69 --- [home2 router]
```

Here is **home1** config (this is my primary router on my cable connection I'd like things to prefer):

```ini
[Match]
Name=vlan69

[Network]
Description=Main user VLAN
DHCP=no
Address=10.6.9.2/24
Address=fd00:1::2/64
Address=fe80::2/64
DHCPServer=no
IPForward=yes
DHCPv6PrefixDelegation=yes
IPv6AcceptRA=no
IPv6SendRA=yes
LLDP=yes
EmitLLDP=nearest-bridge

[Route]
Destination=0.0.0.0/0
Gateway=10.6.9.3
Metric=1069

[Route]
Destination=::/0
Gateway=fe80::3
Metric=1469

[IPv6SendRA]
Managed=no
RouterLifetimeSec=30
RouterPreference=high
OtherInformation=yes
EmitDNS=yes
DNS=fd00:68::69
EmitDomains=yes
Domains=cooperlees.com
DNSLifetimeSec=600
```

Do I have any bad config here that is causing the second RA? Is there something else on the router that could be sending these RAs? I thought the kernel has no way to send an RA and I don't have `radvd` or any other RA daemon running ...

I don't have accept_ra set on the vlan69 interface either
```
root@home2:~# sysctl -a | grep 'vlan69.accept_ra ='
net.ipv6.conf.vlan69.accept_ra = 0
```

Any thoughts here?

### Unexpected behaviour you saw

I see the RA with correct priorites from each router + DNS etc. and then a second RA with medium preference missing the ``OtherInformation`' from each router

home2:

```
 00:00:03.071491 IP6 (flowlabel 0x09d75, hlim 255, next-header ICMPv6 (58) payload length: 104) fe80::3 > ip6-allnodes.cooperlees.com: [icmp6 sum ok] ICMP6, router advertisement, length 104
	hop limit 0, Flags [other stateful], pref low, router lifetime 30s, reachable time 0ms, retrans timer 0ms
	  source link-address option (1), length 8 (1): 00:e0:67:20:1f:c9
	    0x0000:  00e0 6720 1fc9
	  prefix info option (3), length 32 (4): fd00:1::/64, Flags [onlink, auto], valid time 3600s, pref. time 1800s
	    0x0000:  40c0 0000 0e10 0000 0708 0000 0000 fd00
	    0x0010:  0001 0000 0000 0000 0000 0000 0000
	  rdnss option (25), length 24 (3):  lifetime 600s, addr: fd00:70::69
	    0x0000:  0000 0000 0258 fd00 0070 0000 0000 0000
	    0x0010:  0000 0000 0069
	  dnssl option (31), length 24 (3):  lifetime 600s, domain(s): cooperlees.com.
	    0x0000:  0000 0000 0258 0a63 6f6f 7065 726c 6565
	    0x0010:  7303 636f 6d00
 00:00:06.755082 IP6 (flowlabel 0x09d75, hlim 255, next-header ICMPv6 (58) payload length: 56) fe80::3 > ip6-allnodes.cooperlees.com: [icmp6 sum ok] ICMP6, router advertisement, length 56
	hop limit 64, Flags [none], pref medium, router lifetime 30s, reachable time 0ms, retrans timer 0ms
	  prefix info option (3), length 32 (4): fd00:1::/64, Flags [onlink, auto], valid time 2592000s, pref. time 604800s
	    0x0000:  40c0 0027 8d00 0009 3a80 0000 0000 fd00
	    0x0010:  0001 0000 0000 0000 0000 0000 0000
	  source link-address option (1), length 8 (1): 00:e0:67:20:1f:c9
	    0x0000:  00e0 6720 1fc9
```

Home1:
```
 00:00:10.444535 IP6 (flowlabel 0x99898, hlim 255, next-header ICMPv6 (58) payload length: 88) fe80::2 > ip6-allnodes.cooperlees.com: [icmp6 sum ok] ICMP6, router advertisement, length 88
	hop limit 64, Flags [none], pref medium, router lifetime 30s, reachable time 0ms, retrans timer 0ms
	  prefix info option (3), length 32 (4): 2600:6c4e:227f:4c02::/64, Flags [onlink, auto], valid time 2592000s, pref. time 604800s
	    0x0000:  40c0 0027 8d00 0009 3a80 0000 0000 2600
	    0x0010:  6c4e 227f 4c02 0000 0000 0000 0000
	  prefix info option (3), length 32 (4): fd00:1::/64, Flags [onlink, auto], valid time 2592000s, pref. time 604800s
	    0x0000:  40c0 0027 8d00 0009 3a80 0000 0000 fd00
	    0x0010:  0001 0000 0000 0000 0000 0000 0000
	  source link-address option (1), length 8 (1): ac:1f:6b:6f:0d:97
	    0x0000:  ac1f 6b6f 0d97
 00:00:01.233969 IP6 (flowlabel 0x99898, hlim 255, next-header ICMPv6 (58) payload length: 104) fe80::2 > ip6-allnodes.cooperlees.com: [icmp6 sum ok] ICMP6, router advertisement, length 104
	hop limit 0, Flags [other stateful], pref high, router lifetime 30s, reachable time 0ms, retrans timer 0ms
	  source link-address option (1), length 8 (1): ac:1f:6b:6f:0d:97
	    0x0000:  ac1f 6b6f 0d97
	  prefix info option (3), length 32 (4): 2600:6c4e:227f:4c02::/64, Flags [onlink, auto], valid time 3600s, pref. time 1800s
	    0x0000:  40c0 0000 0e10 0000 0708 0000 0000 2600
	    0x0010:  6c4e 227f 4c02 0000 0000 0000 0000
	  rdnss option (25), length 24 (3):  lifetime 600s, addr: pi.hole
	    0x0000:  0000 0000 0258 fd00 0068 0000 0000 0000
	    0x0010:  0000 0000 0069
	  dnssl option (31), length 24 (3):  lifetime 600s, domain(s): cooperlees.com.
	    0x0000:  0000 0000 0258 0a63 6f6f 7065 726c 6565
	    0x0010:  7303 636f 6d00
```

### Steps to reproduce the problem

Apply this config to an interface shown above (maybe with a second networkd instance).

**Comment by yuwata** [comment]
Please provide debugging logs of networkd.

**Comment by cooperlees** [comment]
Sure. Here is the debug logs from home2 - I have them enabled there all the time. I can get home1 if you wish too if they helped. Just enabled them.

[cooper_home2_networkd_debug.log.gz](https://github.com/systemd/systemd/files/11712827/cooper_home2_networkd_debug.log.gz)

**Comment by yuwata** [comment]
Already fixed by 6f812d289946a3e0e386378263b40d09e125752e (v253).
